lib/sqlalchemy/sql/elements.py

```diff
@@ -2711,8 +2756,14 @@ def append(self, clause):
     def _from_objects(self) -> List[FromClause]:
         return list(itertools.chain(*[c._from_objects for c in self.clauses]))
 
-    def self_group(self, against=None):
-        if self.group and operators.is_precedent(self.operator, against):
+    def self_group(
+        self, against: Optional[OperatorType] = None
+    ) -> ClauseElement:
+        if (
+            against
```

Comment:
Some of the `self_group()` overrides require `against` argument because of a call to `is_precedent()`, which conflicts with `Optional` arg annotation - added `against` to the conditional to deal with that without changing the function signature.